component/src/main/java/org/wso2/extension/siddhi/store/ignite/ApacheIgniteConditionVisitor.java

```diff
@@ -0,0 +1,311 @@
+                break;
+            case GREATER_THAN:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_GREATER_THAN);
+                break;
+            case GREATER_THAN_EQUAL:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_GREATER_THAN_EQUAL);
+                break;
+            case LESS_THAN:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_LESS_THAN);
+                break;
+            case LESS_THAN_EQUAL:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_LESS_THAN_EQUAL);
+                break;
+            case NOT_EQUAL:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_NOT_EQUAL);
+                break;
+        }
+        condition.append(ApacheIgniteConstants.WHITESPACE);
+    }
+
+    @Override
+    public void endVisitCompareRightOperand(Compare.Operator operator) { }
+
+    @Override
+    public void beginVisitIsNull(String streamId) { }
+
+    @Override
+    public void endVisitIsNull(String streamId) {
+        condition.append(ApacheIgniteConstants.SQL_IS_NULL).append(ApacheIgniteConstants.WHITESPACE);
+    }
+
+    @Override
+    public void beginVisitIn(String storeId) {
+        condition.append(ApacheIgniteConstants.SQL_IN).append(ApacheIgniteConstants.WHITESPACE);
+    }
+
+    @Override
+    public void endVisitIn(String storeId) { }
+
+    @Override
+    public void beginVisitConstant(Object value, Attribute.Type type) {
+        String name = this.generateConstantName();
+        this.placeholdersConstant.put(name, value);
+        condition.append("[").append(name).append("]").append(ApacheIgniteConstants.WHITESPACE);
+    }
+
+    @Override
+    public void endVisitConstant(Object value, Attribute.Type type) { }
+
+    @Override
+    public void beginVisitMath(MathOperator mathOperator) {
+        condition.append(ApacheIgniteConstants.OPEN_PARENTHESIS);
+    }
+
+    @Override
+    public void endVisitMath(MathOperator mathOperator) {
+        condition.append(ApacheIgniteConstants.CLOSE_PARENTHESIS);
+    }
+
+    @Override
+    public void beginVisitMathLeftOperand(MathOperator mathOperator) { }
+
+    @Override
+    public void endVisitMathLeftOperand(MathOperator mathOperator) { }
+
+    @Override
+    public void beginVisitMathRightOperand(MathOperator mathOperator) {
+        switch (mathOperator) {
+            case ADD:
+                condition.append(ApacheIgniteConstants.SQL_MATH_ADD);
+                break;
+            case DIVIDE:
+                condition.append(ApacheIgniteConstants.SQL_MATH_DIVIDE);
+                break;
+            case MOD:
+                condition.append(ApacheIgniteConstants.SQL_MATH_MOD);
+                break;
+            case MULTIPLY:
+                condition.append(ApacheIgniteConstants.SQL_MATH_MULTIPLY);
+                break;
+            case SUBTRACT:
+                condition.append(ApacheIgniteConstants.SQL_MATH_SUBTRACT);
+                break;
```

Comment:
add default case and throw an exception if other unsupported cases can be passed into this function